File: 05-02-binding-adapter/java-05-02-binding-adapter/app/src/main/java/com/example/bindingadapter/MainActivity.java
package com.example.bindingadapter;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ObservableField;

import com.example.dalinaum.bindingadapter.R;
import com.example.dalinaum.bindingadapter.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        ViewModel viewModel = new ViewModel();
        binding.setViewModel(viewModel);
        viewModel.title.set("Reverse String");
    }

    public static class ViewModel {
        public ObservableField<String> title = new ObservableField<>();
    }
}

File: 05-02-binding-adapter/kotlin-05-02-binding-adapter/app/src/main/java/com/example/bindingadapter/MainActivity.kt
package com.example.bindingadapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ObservableField
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

import com.example.dalinaum.bindingadapter.R
import com.example.dalinaum.bindingadapter.databinding.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        val viewModel = ViewModel()
        binding.viewModel = viewModel
        viewModel.title.set("Reverse String")
    }

    class ViewModel {
        var title = ObservableField<String>()
    }
}
